tests/commands/envelope_tests.py

```diff
@@ -0,0 +1,103 @@
+# encoding=utf-8
+# Copyright © 2017 Dylan Baker
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the alot.commands.envelope module."""
+
+from __future__ import absolute_import
+import os
+import contextlib
+import shutil
+import tempfile
+import unittest
+
+import mock
+
+from alot.commands import envelope
+
+
+@contextlib.contextmanager
+def temporary_directory(suffix='', prefix='', dir=None):
+    """Python3 interface implementation.
+
+    Python3 provides a class that can be used as a context manager, which
+    creates a temporary directory and removes it when the context manager
+    exits. This function emulates enough of the interface of
+    TemporaryDirectory, for this module to use, and is designed as a drop in
+    replacement that can be replaced after the python3 port.
+
+    The only user visible difference is that this does not implement the
+    cleanup method that TemporaryDirectory does.
+    """
+    directory = tempfile.mkdtemp(suffix=suffix, prefix=prefix, dir=dir)
+    yield directory
+    shutil.rmtree(directory)
+
+
+class TestAttachCommand(unittest.TestCase):
+    """Tests for the AttachCommaned class."""
+
+    def test_single_path(self):
+        """A test for an existing single path."""
+        ui = mock.Mock()
+
+        with temporary_directory() as d:
+            testfile = os.path.join(d, 'foo')
+            with open(testfile, 'w') as f:
+                f.write('foo')
+
+            cmd = envelope.AttachCommand(path=testfile)
+            cmd.apply(ui)
+        ui.current_buffer.envelope.attach.assert_called_with(testfile)
+
+    def test_user(self):
+        """A test for an existing single path prefaced with ~/."""
```

Comment:
I removed a mock patch from that test that I thought I needed when I started, then realized I didn't. I think I somehow backed out the change to make testfile = '~/foo' when I did that. I'll fix that.